### ISSUE TITLE
GNOME freezes on MacBook Pro

### ISSUE THREAD

**Issue by 0xA1B2** [clearlinux/distribution#1524]
Hello,

I use a MacBook Pro 2011 MC725LL/A with a fresh install of CL and VMware Workstation on top.
I happened now twice that the GNOME desktop freezes, the clock does not move forward.
The mouse cursors still moves, and the VM is still running as it serves a website.

The freeze happened when I started Chrome, and some hours before it happened when I tried to open the VMware Workstation GUI. No graphics drivers are installed other than CL defaults.

x0vncserver and a node.js process running (PM2) on CL are not responding.

Where are the logs to analyse this issue ? Thanks.

**Comment by lebensterben** [comment]
>Hello, ...

@0xA1B2 I had similar issue on a desktop. Can you go to tty? When freeze occurs, I cannot.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by 0xA1B2** [comment]
Sadly not. When the freeze happens, the GUI is kind of locked. The mouse also was smooth at first, but then became sluggish. VNC terminated, node.js stopped and it's not possible to do any input.

Funny still, the VMware VM operated normal, PM2 web monitoring from inside the (Debian) VM showed an increase in the event loop time, but not abnormal. The PM2 node.js process on CL directly died lol

Today I after the second crash experience I let the notebook sit in its misery as I had to drive off, and actually GNOME came back to life, I could close Chrome now which finally loaded fully and work with the desktop, still slow, far from responsive, but after 3-5 seconds I got a response from the GUI to act on a mouse click. 

Nice actually that there was some recovery, just still things like that should be investigated and fixed if possible. Maybe the old hardware from 2011 needs some stuff that the pretty new CL does not have.

On the other side, I use a Lenovo T520 as my workhorse with CL on it, also 2011 generation, same CPU actually as the MacBook Pro, and its just fast with no limit on the T520. It seems that CL struggles a bit on Apple hardware, or Apple did some odd things and CL has a hard time to stay alive.

**Comment by lebensterben** [comment]
>Sadly not. When the freeze happens, the GUI is kind of locked. The mouse was smooth at first, but then became sluggish. VNC terminated, node.js stopped and it's not possible to do any input. Oh deer. ...

@0xA1B2 

As for freeze, my workstation is purchased in 2018 and certainly it is not a legacy build. And in the past I have exact the same experience you described.

For performance: There is another factor that laptops sometimes limit the clock speed to reduce temperature.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by 0xA1B2** [comment]
Yes, My T520 has 16GB RAM compared to Apples 4GB and an Intel SSD vs HDD 5400 at that time. Still giving some respect to old age, GNOME/Wayland/Intel GPU driver, should be in RAM anyways. 

After CL install at the MacBook Pro it felt sluggish, a delay of 1-2 seconds to get things loaded vs Lenovo which is instant. I guess that's the RAM/HDD deficit and not being used to that old stuff anymore but I can accept that.

However, the old specs should not freeze GNOME desktop to a stall. There was enuff RAM available, and 64GB of swap space for urgent needs. 

I'm not a Linux Pro at all, more of lower medium to be honest, but it seems that on the MacBook there is some kind of a process fight going on, not taking much CPU power to be noticed, but enuff to stall GNOME to become useless. 

I sense a Wayland/AMD<>Intel GPU bug here, and if given instructions, I'm more than happy to help.

**Comment by lebensterben** [comment]
Check this https://www.ifixit.com/Guide/MacBook+Pro+13-Inch+Unibody+Early+2011+Heat+Sink+Replacement/5153

Unibody MacBook are notoriously sluggish when it's old. I guess you had not opened your MacBook, or otherwise you'd be astonished at the dust in it.

**Comment by bktan8** [comment]
@0xA1B2  - Trying to reproduce your issue.  But need some clarifications...
Did you install CL on baremetal/on the Mac or you're running CL as VM using VMware Workstation as the hypervisor?  If the latter, which version of MacOS X are you running?  And which version of VMWare Workstation?
